common/flatpak-dir.c

```diff
@@ -9494,6 +9494,32 @@ flatpak_dir_find_remote_related (FlatpakDir *self,
                   continue;
                 }
 
+              /* Don't download anything for "download-if=active-gl-driver"
+               * extension points if there is an unmanaged extension for it
+               * installed already */
```

Comment:
I think it doesn't necessarily make sense to have this logic be specific to unmanaged extensions. If our intended behavior is that, for example, the highest priority available driver is the best, it should be the same whether that extension is unmanaged or a regular one.

The only special-casing of the host extension in that case is that it would(/could/should/...) appear first in the drivers list, but because it's never available for download, stops any others being retrieved if it's available. I guess we'd also need to put it first in the list. :) But that doesn't seem too controversial to me.